pkg/registry/registry.go

```diff
@@ -1,20 +1,28 @@
 package registry
 
+import "time"
+
 // Registry implements a generic interface for service discovery
 type Registry interface {
-	// Register a service with the Registry with a given name
+	// Register a node with the Registry with a given name
 	// Returns a Registration id or error
-	Register(string) (string, error)
+	Register(string, time.Duration) (string, error)
 
 	// Deregister a service which was registred with a id
 	// Returns error on failure
 	Deregister(string) error
 
+	// IsRegistered checks is node with registration_id is registred with registry
+	IsRegistered(string) bool
+
+	// Renew a regestration using registration_id
+	Renew(string) error
+
 	// Acquire a lock for a registration_id on a given key and value pair
-	Acquire(string, string, string) (bool, error)
+	Acquire(string, string, string) bool
 
 	// Release a lock for a restration_id on a given key and value pair
-	Release(string, string, string) (bool, error)
+	Release(string, string, string) bool
```

Comment:
error may still need to be handled differently?

Reply:
currently it was getting handled same way so removed, can add back if we need different handling